### PR TITLE
added basic battle featrues

### LINKED ISSUE #18: Battle: 1st iteration
We need to begin implementing our battling system. Since this will probably be our most complex mechanic, it's going to be done in iterations.
For the first iteration, we are going to have a simple, pre-determined battle sequence between the main character and a creature.
### ACCEPTANCE CRITERIA:

- Create a function that starts a battle sequence. The beginning will be marked by an appropriate output to the console (something like "Battle starts!")
- The function will create a hand-written "goblin" creature, and a main character with default stats
- It will print the stats of the two foes to the console
- It will compare their HP, and decide the winner based on the highest one
- Finally, it will print out the winner to the console, along with an appropriate message to end the battle (something like "Battle ends!")
- Have this function be called at the end of our main game.

### PR DESCRIPTION
**Pull request by DIONYSUS-BASOUNAS**
closes #18

### CODE REVIEW

File: src/battle/battle.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include "battle.h"
+#include "../utils/utils.h"
+#include "../creature/creature.h"
+#include "../creature/main_character/main_character.h"
+#include <iostream>
+
+void Battle()
+{
+	std::cout << "Battle starts!" << std::endl;
+	Creature Goblin(10, 14, 35, 16, 14, 10, 8, 7, 12, "Goblin");
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:10]: goblin is a variable, so it should start with a lower-case 'g'

File: src/battle/battle.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		std::cout
+			<< "The Goblin has more HP than the main character!, Goblin wins!"
+			<< std::endl;
+		}
+	else if (Goblin.GetHP() < main_character.GetHP())
+	{
+		std::cout << "The main character has more HP than the Goblin! You win!"
+				  << std::endl;
+		std::cout << "Battle ends" << std::endl;
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:24]: This should be outside the 'if' statements, in order to apply to both of them

File: src/battle/battle.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	}
+}
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:26]: Empty line at the EOF

File: src/battle/battle.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	Goblin.PrintStats();
+	MainCharacter main_character("Traveler");
+	main_character.PrintStats();
+	if (Goblin.GetHP() >= main_character.GetHP())
+	{
+		std::cout
+			<< "The Goblin has more HP than the main character!, Goblin wins!"
+			<< std::endl;
+		}
+	else if (Goblin.GetHP() < main_character.GetHP())
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:20]: Should we consider having a third statement for when their HPs are  equal?

File: src/battle/battle.h [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+/**
+ * @brief A function that takes you to battle
+ */
+void Battle();
```
Comment (achilleasmitos) [inline src/battle/battle.h:5]: Empty line at the EOF

File: src/battle/battle.h [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+/**
+ * @brief A function that takes you to battle
```
Comment (achilleasmitos) [inline src/battle/battle.h:3]: Please be a bit more descriptive. With these comments, one still needs to go to the implementation of the function to see what it does. A simple "it starts a static battle between a main character with default stats and a hand-written goblin creature, and compares their HP to decide the winner" should be enough for this iteration.

File: src/creature/creature.h [hunk 7]
```diff
@@ -21,6 +21,7 @@ class Creature
 		std::string creature_type);
 
 	void PrintStats();
+	int GetHP();
```
Comment (achilleasmitos) [inline src/creature/creature.h:24]: Empty line above. Also, getters and setters should follow the variable naming convention (so it should be written as `get_hp()`)

File: src/creature/creature.cpp [hunk 8]
```diff
@@ -38,3 +38,4 @@ void Creature::PrintStats()
 	std::cout << "Creature type: " << m_creature_type << std::endl;
 	std::cout << "==============================" << std::endl;
 }
+int Creature::GetHP() { return m_hp; }
```
Comment (achilleasmitos) [inline src/creature/creature.cpp:41]: Empty line above

File: src/creature/main_character/main_character.cpp [hunk 9]
```diff
@@ -81,3 +81,4 @@ bool MainCharacter::CheckForAlphaOrWhitespace(std::string string)
 		return true;
 	}
 }
+int MainCharacter::GetHP() { return m_hp; }
```
Comment (achilleasmitos) [inline src/creature/main_character/main_character.cpp:84]: Ditto (do the same as in Creature)

File: src/creature/main_character/main_character.h [hunk 10]
```diff
@@ -30,10 +30,10 @@ class MainCharacter : public Creature
 
 	std::string get_name();
 	void set_name(std::string name);
+	int GetHP();
```
Comment (achilleasmitos) [inline src/creature/main_character/main_character.h:33]: Ditto (do the same as in Creature)

File: src/creature/main_character/main_character.h [hunk 11]
```diff
@@ -30,10 +30,10 @@ class MainCharacter : public Creature
...
 
 private:
 	std::string m_name;
-
```
Comment (achilleasmitos) [inline src/creature/main_character/main_character.h:36]: Why remove the empty line here?

**Review by achilleasmitos — COMMENTED** [review]
Lots of comment, but none indicate errors. Nice job!

File: src/battle/battle.cpp (hunk 1 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:10]: done

File: src/battle/battle.cpp (hunk 2 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:24]: done

File: src/battle/battle.cpp (hunk 3 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:26]: done

File: src/battle/battle.cpp (hunk 4 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:20]: i had it at first, but like in d&d you have to go over. and i dosen't make sense to have to nothing or both lose if they are equal

File: src/battle/battle.h (hunk 6 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.h:3]: done

File: src/creature/creature.h (hunk 7 above)
Comment (DIONYSUS-BASOUNAS) [reply src/creature/creature.h:24]: done

File: src/creature/main_character/main_character.cpp (hunk 9 above)
Comment (DIONYSUS-BASOUNAS) [reply src/creature/main_character/main_character.cpp:84]: done

File: src/creature/creature.cpp (hunk 8 above)
Comment (DIONYSUS-BASOUNAS) [reply src/creature/creature.cpp:41]: done

File: src/creature/main_character/main_character.h (hunk 11 above)
Comment (DIONYSUS-BASOUNAS) [reply src/creature/main_character/main_character.h:36]: it was a miss input

File: src/creature/main_character/main_character.h (hunk 10 above)
Comment (DIONYSUS-BASOUNAS) [reply src/creature/main_character/main_character.h:33]: done

File: src/battle/battle.cpp (hunk 4 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:20]: would you like me to add it back?

**Review by achilleasmitos — APPROVED** [review]

**Pull request merged**